<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>KeyGenDialog</class>
 <widget class="QDialog" name="KeyGenDialog">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>400</width>
    <height>300</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>生成SSH密钥</string>
  </property>
  <property name="minimumWidth">
   <number>400</number>
  </property>
  <layout class="QVBoxLayout" name="verticalLayout">
   <item>
    <widget class="QLabel" name="infoLabel">
     <property name="text">
      <string>生成新的SSH密钥对。密钥将保存在您的用户目录下的.ssh文件夹中。</string>
     </property>
     <property name="wordWrap">
      <bool>true</bool>
     </property>
    </widget>
   </item>
   <item>
    <layout class="QHBoxLayout" name="emailLayout">
     <item>
      <widget class="QLabel" name="emailLabel">
       <property name="text">
        <string>邮箱地址：</string>
       </property>
      </widget>
     </item>
     <item>
      <widget class="QLineEdit" name="emailEdit">
       <property name="placeholderText">
        <string>your_email@example.com</string>
       </property>
      </widget>
     </item>
    </layout>
   </item>
   <item>
    <layout class="QHBoxLayout" name="typeLayout">
     <item>
      <widget class="QLabel" name="typeLabel">
       <property name="text">
        <string>密钥类型：</string>
       </property>
      </widget>
     </item>
     <item>
      <widget class="QComboBox" name="typeCombo">
       <item>
        <property name="text">
         <string>RSA (推荐)</string>
        </property>
        <property name="userData">
         <string>rsa</string>
        </property>
       </item>
       <item>
        <property name="text">
         <string>ED25519</string>
        </property>
        <property name="userData">
         <string>ed25519</string>
        </property>
       </item>
       <item>
        <property name="text">
         <string>ECDSA</string>
        </property>
        <property name="userData">
         <string>ecdsa</string>
        </property>
       </item>
      </widget>
     </item>
    </layout>
   </item>
   <item>
    <layout class="QHBoxLayout" name="bitsLayout">
     <item>
      <widget class="QLabel" name="bitsLabel">
       <property name="text">
        <string>密钥长度：</string>
       </property>
      </widget>
     </item>
     <item>
      <widget class="QComboBox" name="bitsCombo">
       <item>
        <property name="text">
         <string>4096位 (更安全)</string>
        </property>
        <property name="userData">
         <number>4096</number>
        </property>
       </item>
       <item>
        <property name="text">
         <string>2048位 (标准)</string>
        </property>
        <property name="userData">
         <number>2048</number>
        </property>
       </item>
      </widget>
     </item>
    </layout>
   </item>
   <item>
    <layout class="QHBoxLayout" name="passLayout">
     <item>
      <widget class="QLabel" name="passLabel">
       <property name="text">
        <string>密钥密码：</string>
       </property>
      </widget>
     </item>
     <item>
      <widget class="QLineEdit" name="passEdit">
       <property name="echoMode">
        <enum>QLineEdit::Password</enum>
       </property>
       <property name="placeholderText">
        <string>可选，留空表示无密码</string>
       </property>
      </widget>
     </item>
    </layout>
   </item>
   <item>
    <layout class="QHBoxLayout" name="confirmLayout">
     <item>
      <widget class="QLabel" name="confirmLabel">
       <property name="text">
        <string>确认密码：</string>
       </property>
      </widget>
     </item>
     <item>
      <widget class="QLineEdit" name="confirmEdit">
       <property name="echoMode">
        <enum>QLineEdit::Password</enum>
       </property>
       <property name="placeholderText">
        <string>再次输入密码</string>
       </property>
      </widget>
     </item>
    </layout>
   </item>
   <item>
    <widget class="QDialogButtonBox" name="buttonBox">
     <property name="orientation">
      <enum>Qt::Horizontal</enum>
     </property>
     <property name="standardButtons">
      <set>QDialogButtonBox::Cancel|QDialogButtonBox::Ok</set>
     </property>
    </widget>
   </item>
  </layout>
 </widget>
 <resources/>
 <connections>
  <connection>
   <sender>buttonBox</sender>
   <signal>accepted()</signal>
   <receiver>KeyGenDialog</receiver>
   <slot>accept()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>248</x>
     <y>254</y>
    </hint>
    <hint type="destinationlabel">
     <x>157</x>
     <y>274</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>buttonBox</sender>
   <signal>rejected()</signal>
   <receiver>KeyGenDialog</receiver>
   <slot>reject()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>316</x>
     <y>260</y>
    </hint>
    <hint type="destinationlabel">
     <x>286</x>
     <y>274</y>
    </hint>
   </hints>
  </connection>
 </connections>
</ui>